changelog/11256.improvement.md

```diff
@@ -0,0 +1,2 @@
+Implements a new subclass of `LockStore` entitled `ConcurrentRedisLockStore` that handles concurrent scenarios when separate Rasa instances are trying to acquire the lock at the same time.
+A deprecation warning was issued for `RedisLockStore` which will be removed in version `4.0.0` in favour of the new concurrent implementation.
```

Comment:
```suggestion
A deprecation warning will be issued for the usage of the prior `RedisLockStore`. It will be removed in version `4.0.0` in favour of the new concurrent implementation.
```